File: Examples/Node.js/SDK/workbook/ConvertWorkbookToAnotherFormatWithoutStorage.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var CellsApi = require('asposecellscloud');
var configProps = require('../Data/config.json');
var data_path = '../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Cells API SDK
var cellsApi = new CellsApi(config);

// Set input file name
var filename = "Sample_Test_Book";
var name = filename +  ".xls";
var format = "pdf";
try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');
	
	// Invoke Aspose.Cells Cloud SDK API to convert workbook to different file formats without storage
	cellsApi.PutConvertWorkBook(format, null, null, data_path + name, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			
			// Download Workbook from response stream
			assert.equal(responseMessage.status, 'OK');
			var outfilename = filename + "_out." + format;
			var writeStream = fs.createWriteStream(data_path + outfilename);
			writeStream.write(responseMessage.body);

			});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Android/app/src/main/java/com/aspose/cells/cloud/examples/workbook/ConvertWorkbookToAnotherFormatWithoutStorage.java
package com.aspose.cells.cloud.examples.workbook;


import com.aspose.cells.api.CellsApi;
import com.aspose.cells.cloud.examples.Configuration;
import com.aspose.cells.cloud.examples.Utils;
import com.aspose.cells.model.ResponseMessage;
import com.aspose.storage.api.StorageApi;

import java.io.IOException;
import java.io.File;
import com.aspose.cells.cloud.examples.R;
import android.content.Context;


public class ConvertWorkbookToAnotherFormatWithoutStorage {

	public static void execute(Context context) throws IOException {

		//ExStart: convert-workbook-to-different-format-without-storage
		try {
			File inputFile = Utils.stream2file("sample1","xlsx", context.getResources().openRawResource(R.raw.sample1));
			File outputFile = Utils.stream2file("sample2","xlsx", context.getResources().openRawResource(R.raw.sample2));

			// Instantiate Aspose Storage API SDK
			StorageApi storageApi = new StorageApi(Configuration.apiKey, Configuration.appSID, true);

			// Instantiate Aspose Words API SDK
			CellsApi cellsApi = new CellsApi(Configuration.apiKey, Configuration.appSID, true);
			String format = "pdf";
			String password = "";
			String outPath = "";

			String xml = "<PdfSaveOptions>" + "<SaveFormat>pdf</SaveFormat>" + "<FileName>"
					+ "sample1.xlsx" + "</FileName>" + "<ImageCompression>Jpeg</ImageCompression>"
					+ "<JpegQuality>70</JpegQuality>" + "<TextCompression>Flate</TextCompression>"
					+ "</PdfSaveOptions>";

			ResponseMessage cr = cellsApi.PutConvertWorkBook(format, password, outPath,inputFile,
					xml.getBytes("UTF-8"));

			Utils.copyInputStreamToFile(cr.getInputStream(),outputFile);

		} catch (Exception e) {
			e.printStackTrace();
		}
		//ExEnd: convert-workbook-to-different-format-without-storage
	}
}